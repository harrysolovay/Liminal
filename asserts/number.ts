import { assert } from "../util/mod.ts"

export function min(value: number, min: number) {
  assert(value >= min, `Must be gte ${min}; received ${value}.`)
}

export function max(value: number, max: number) {
  assert(value <= max, `Must be lte ${max}; received ${value}.`)
}
