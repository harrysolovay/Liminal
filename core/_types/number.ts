import type { Type } from "../Type.ts"
import { declareType } from "./_declareType.ts"

export const number: Type<number> = declareType(() => number)
