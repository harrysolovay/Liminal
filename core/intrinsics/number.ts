import { declareType } from "../declareType.ts"
import type { Type } from "../Type.ts"

export const number: Type<number> = declareType({
  type: "number",
  self() {
    return number
  },
})
