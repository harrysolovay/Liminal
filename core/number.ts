import type { Type } from "../Type.ts"
import { declare } from "../TypeDeclaration.ts"
import { assert } from "../util/assert.ts"

export const number: Type.Initial<number, {
  min: number
  max: number
  int: boolean
}> = declare<number>()({
  name: "number",
  source: {
    getType: () => number,
  },
  subschema: (_0, ctx) => ({
    type: ctx.refinements.int ? "integer" : "number",
  }),
  assertRefinementsValid: ({ min, max }) =>
    assert(
      !(typeof min === "number" && typeof max === "number") || min <= max,
      "Refinement min is greater than its max.",
    ),
  assertRefinements: {
    min: (value, min) => assert(value >= min, `value >= ${min}`),
    max: (value, max) => assert(value <= max, `value <= ${max}`),
  },
})
