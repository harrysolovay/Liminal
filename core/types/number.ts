import { Type } from "../Type.ts"

export const number: Type<number> = Type({
  name: "number",
  getAtom: () => number,
})
