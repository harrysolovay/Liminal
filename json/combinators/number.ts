import { Type } from "../../core/mod.ts"

export const number: Type<number> = Type({
  getType: () => number,
})
