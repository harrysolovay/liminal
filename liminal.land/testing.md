# Liminal Testing <Badge type="warning" text="beta" />
