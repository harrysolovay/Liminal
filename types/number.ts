import { declareType, type Type } from "../core/mod.ts"

export const number: Type<number> = declareType({
  name: "number",
  source: {
    getType: () => number,
  },
})
