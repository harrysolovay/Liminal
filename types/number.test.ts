import { assertTypeSnapshot } from "testing"
import { number } from "./number.ts"

Deno.test("number", async (t) => {
  await assertTypeSnapshot(t, number)
})
