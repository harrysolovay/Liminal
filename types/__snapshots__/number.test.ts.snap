export const snapshot = {};

snapshot[`number 1`] = `
{
  type: "number",
}
`;

snapshot[`number 2`] = `
{
  description: "One.",
  type: "number",
}
`;

snapshot[`number 3`] = `
{
  description: "One. Two.",
  type: "number",
}
`;
